lib/ranked-model/ranker.rb

```diff
@@ -69,6 +69,16 @@ def update_rank! value
           update_all(ranker.column => value)
       end
 
+      def rank_after(record)
+        # Add check for equality of record and current_ranker
```

Comment:
This still needs to be added? Add `def ==` to the ranker class that checks that `[ :scope, :with_same, :unless ]` are the same. Then you can just raise if `rank != self`. It doesn't read very nice though, mainly due to the variable name. I'm keen for other ideas on the method name :)

Reply:
Actually, I don't think it is required anymore. The only method that is calling this is `#{ranker.name}_rank_after!` which creates the ranker for the `record` param.

Reply:
That's true, though this method could be called by the end user? Is there a way to protect it?

Reply:
I think for the end user to call this method they would need to know that the method `rank_after` exists at the ranker level.
But yea, this could be called by the end user. 

Another related question which I am pondering on - 

Does this method require to be the same ranker?
Technically it could be any ranker till it has the interface `relative_rank`

Reply:
Correct, it would just have strange results because the position would be different in a different rank. This has made me think about the interface in general. Is `pond_rank_after!` a nice interface. I wonder if it would be better to have an intermediate object (which we already have in the ranker): `pond_ranker.rank_after(record)`. This is especially something to consider as we begin adding more methods to the ranker. It also guarantees that when we pass a record into that method, that it gets wrapped in the same ranker as the caller.

What do you think? We should probably change `relative_rank` too?

Reply:
I am not sure if I understood the above correctly 😓😓 

So, currently the method exists inside the ranker itself and the args to the method are is a ranker object.

I don't think we should expose `ranker` class interfaces directly to the user.

The user should just be able to call - `duck.pond_rank_after!(another_duck)`
Otherwise, they will have to create the ranker class objects and then call. 

The method `#{ranker.name}_rank_after!` accepts the record and converts the record into a ranker record and then calls the `rank_after` interface on the ranker object.

Reply:
Sorry about that. I was trying to widen the conversation to look at the interface we're providing the end user. It seems we're cluttering the model class with methods that might be better hanging off an associated class. It hasn't been an issue until now because there was only one method used for positioning `#{rank_name}_position=()`. Now we're moving away from that I wonder if we should clean up this concept and provide a rank object to the end user off of which we add our methods:

`duck.rank(:pond).position=(position)`
`duck.rank(:pond).position`
`duck.rank(:pond).relative_position`
`duck.rank(:pond).rank_after!(other_duck)`

Before this current work, `.position=` was used as a catchall method to move things around (thus the shortcomings), though it could be used like this for this current PR:

`duck.pond_position = :after, other_duck`

I hope that clarifies things. @mixonic, I know you're well out of this now, but if you had a few seconds to weigh in on this it would be super helpful :) Just on the direction you think we should choose, not the idea of the PR itself.

Reply:
Hi @brendon,

First of all I'm extremely sorry for abandoning the PR for so long 😞 

I think with new methods getting added. I can imagine similar `rank_before!` method as well to the wider rank interface.

I'm thinking of something like - 

```
def rank(rank_name)
  ranker(rank_name) # Give the ranker instance
end

module RankedModel
  class Ranker
    def initialize()
    end

    def rank_after!()
    end
  end
end

duck.rank(:pond).rank_after!(other_duck)
duck.rank(:pond).position
```

I think the above is definitely a better way to handle calling. If a duck has more than 5 different kind of ranks, it becomes even more confusing to call specific methods with the existing interface.

I think we could create a new issue to implement this and just to discuss the interface and other doubts which I've - would be support existing interface, how would it work along with the existing interface? 🤔 

Meanwhile - if you want we can go ahead with this and think about the new interface in a separate issue :)